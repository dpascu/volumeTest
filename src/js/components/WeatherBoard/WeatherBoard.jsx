import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Loader from '../Loader';

/**
 * Board for displaying the weather
 * @param {Object} props
 */
const WeatherBoard = ({ display, temperature, location, icon }) => {
  const tempDisplay = temperature.toFixed(1);
  return (
    <div className="weather">
      {display
        ? (
          <Fragment>
            <div className="location">
              {location}
            </div>
            <div>
              {icon && <img src={`http://openweathermap.org/img/w/${icon}.png`} alt="icon" />}
              <span className="temperature">
                {`${tempDisplay}ยบ`}
              </span>
            </div>
          </Fragment>
        )
        : <Loader />}
    </div>);
};

WeatherBoard.propTypes = {
  display: PropTypes.bool,
  temperature: PropTypes.number.isRequired,
  location: PropTypes.string.isRequired,
  icon: PropTypes.string,
};

WeatherBoard.defaultProps = {
  display: true,
  icon: null,
};

export default WeatherBoard;
