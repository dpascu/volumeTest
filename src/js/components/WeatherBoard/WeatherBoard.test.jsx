/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';

import WeatherBoard from './WeatherBoard';


describe('Component WeatherBoard', () => {
  it('should render component', () => {
    const component = renderer.create(
      <WeatherBoard
        display={false}
        temperature={12}
        location="London"
        icon="000"
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component', () => {
    const component = renderer.create(
      <WeatherBoard
        display
        temperature={12}
        location="London"
        icon="000"
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
