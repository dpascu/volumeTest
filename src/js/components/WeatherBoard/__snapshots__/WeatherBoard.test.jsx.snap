// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component WeatherBoard should render component 1`] = `
<div
  className="weather"
>
  <img
    alt="loading"
    src="../../../img/loader.gif"
  />
</div>
`;

exports[`Component WeatherBoard should render component 2`] = `
<div
  className="weather"
>
  <div
    className="location"
  >
    London
  </div>
  <div>
    <img
      alt="icon"
      src="http://openweathermap.org/img/w/000.png"
    />
    <span
      className="temperature"
    >
      12.0ยบ
    </span>
  </div>
</div>
`;
